### PR TITLE
[aot_inductor] Enable test_aot_inductor tests for ROCm

### PR DESCRIPTION
**Pull request by pragupta**
Fixes #ISSUE_NUMBER

cc @jeffdaily @sunway513 @jithunnair-amd @pruthvistony @ROCmSupport @dllehr-amd @jataylo @hongxiayang @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @peterbell10 @ipiszy @yf225 @chenyang78 @kadeng @muchulee8 @aakhundov @ColinPeppler @amjames @desertfire @chauhang

### CODE REVIEW

File: torch/_inductor/codegen/cpp_wrapper_cuda.py [hunk 1]
```diff
@@ -78,8 +78,8 @@ def write_header(self):
             do {                                               \\
                 CUresult code = EXPR;                          \\
                 const char *msg;                               \\
-                cuGetErrorString(code, &msg);                  \\
-                if (code != CUDA_SUCCESS) {                    \\
+                CUresult err = cuGetErrorString(code, &msg);                  \\
+                if (err != CUDA_SUCCESS || code != CUDA_SUCCESS) {                    \\
```
Comment (aakhundov) [inline torch/_inductor/codegen/cpp_wrapper_cuda.py:82]: If `err != CUDA_SUCCESS` can we assume that `msg` is populated?

Comment (pragupta) [reply torch/_inductor/codegen/cpp_wrapper_cuda.py:82]: As per the documentation, if `err != CUDA_SUCCESS` then msg will be set to NULL.

See: https://docs.nvidia.com/cuda/cuda-driver-api/group__CUDA__ERROR.html#group__CUDA__ERROR_1g72758fcaf05b5c7fac5c25ead9445ada

**Comment by jithunnair-amd** [comment]
@pragupta Please fix lint errors.

**Comment by jithunnair-amd** [comment]
@malfet Straightforward skips, please review and approve this PR, so I can merge if CI is clean

**Review by jithunnair-amd — APPROVED** [review]

**Review by malfet — APPROVED** [review]

**Pull request closed without merging**